doc/external-access.md

```diff
@@ -0,0 +1,117 @@
+# Enable external access
+
+By default, a Pravega cluster uses `ClusterIP` services which are only accessible from within Kubernetes. However, when creating the Pravega cluster resource, you can opt to enable external access.
+
+In Pravega, clients initiate the communication with the Pravega Controller, which is a stateless component frontended by a Kubernetes service that load-balances the requests to the backend pods. Then, clients discover the individual Segment Store instances to which they directly read and write data to. Clients need to be able to reach each and every Segment Store pod in the Pravega cluster.
+
+If your Pravega cluster needs to be consumed by clients from outside Kubernetes (or from another Kubernetes deployment), you can enable external access in two ways, depending on your environment constraints and requirements. Both ways will create one service for all Controllers, and one service for each Segment Store pod.
+
+1. Via [`LoadBalancer`](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer) service type.
+2. Via [`NodePort`](https://kubernetes.io/docs/concepts/services-networking/service/#nodeport) service type.
```

Comment:
Do you think we need to explicitly mention that `NodePort` is not supported in PKS+NSX-T environment? Thanks.

Reply:
No, I don't think so. This operator is not dependent on any particular Kubernetes platform (e.g. GKE, EKS, AKS, PKS...). We shouldn't attempt to enumerate them and keep track of the features they support and the features they don't, especially considering that that's an information that will be changing over time. It's up to the users to investigate what service type is suitable for them in terms of support and use case.

Reply:
Make sense. Thanks!